### PR TITLE
Solution for #1967

### LINKED ISSUE #1967: GCD in JS
Hey! There's no implementation of GCD on C++, can i go ahead with this?
https://github.com/jainaman224/Algo_Ds_Notes/tree/master/Gcd

### PR DESCRIPTION
**Pull request by heisastark**
Fixes #1967

#### Checklist:
- ☑ 4 space indentation.
- ☑ Coding conventions are followed.
- ☑ Input is taken dynamically.
- ☑ Sample Input / Output is added at the end of file.
- ☑ Logic Documentation (Comments).
- ☑ File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
JS file for the problem added in gcd problem folder

#### Languages Used:
JavaScript

#### File Added:
gcd.js 

Thanks!

### CODE REVIEW

**Comment by asha15** [comment]
please spacify the issue number next to `Fixes #`

File: Gcd/gcd.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*
+JS Implementation of GCD Program using Euclidean Algorithm
+*/
+
+function gcd(m, n)
+{
+    if(m<n)
```
Comment (asha15) [inline Gcd/gcd.js]: keep space before and after

**Comment by heisastark** [comment]
@asha15 added 👍

File: Gcd/gcd.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        var temp = m;
+        m = n;
+        n = temp;
+    }
+    if(m%n===0)
```
Comment (asha15) [inline Gcd/gcd.js]: keep space before and after operator

File: Gcd/gcd.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        return n;
+    }
+    else
+    {
+        return gcd(n, m%n);
```
Comment (asha15) [inline Gcd/gcd.js]: please keep space before and after

File: Gcd/gcd.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
```
Comment (asha15) [inline Gcd/gcd.js]: Remove this line

File: Gcd/gcd.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+function gcd(m, n)
+{
+    if(m < n)
+    {
+        var temp = m;
+        m = n;
+        n = temp;
+    }
+    if(m%n === 0)
```
Comment (asha15) [inline Gcd/gcd.js]: keep space before and after %

**Comment by asha15** [comment]
please squash the commit and push it

**Comment by asha15** [comment]
Great job. can you squash your commits and push it
then will go for merge

File: Gcd/gcd.js (hunk 5 above)
Comment (heisastark) [reply Gcd/gcd.js]: Done ma'am

**Comment by heisastark** [comment]
Yes ma'am :)

**Comment by heisastark** [comment]
Ma'am, there were errors with squashing, i created a new PR #2294, It is ready to merge.
Kindly have a look. @asha15

**Pull request closed without merging**